### PR TITLE
Improve test coverage

### PR DESCRIPTION
**Pull request by jhottenstein**
- bump .ruby_version to 2.6.4
- fix simplecov deprecation warning
- use shared examples to share tests between spec/unit/client/active_record and spec/unit/client/redis 
- fix two bugs in redis client classes
- use skipped tests to highlight two potential bugs in redis client classes

### CODE REVIEW

**Comment by jhottenstein** [comment]
Posting now for feedback, I'm going to add some more tests for daemon/apns* classes

File: lib/rpush/client/active_model/wns/notification.rb [hunk 1]
```diff
@@ -23,6 +23,8 @@ def self.included(base)
               validates :uri, presence: true
               validates :uri, format: { with: %r{https?://[\S]+} }
               validates :data, presence: true, unless: :skip_data_validation?
+              validates_with Rpush::Client::ActiveModel::PayloadDataSizeValidator,
+                             limit: 5120
```
Comment (jhottenstein) [inline lib/rpush/client/active_model/wns/notification.rb]: I assume we would want this: https://docs.oracle.com/html/E59511_02/api/platforms/wns/

File: lib/rpush/client/redis/pushy/notification.rb [hunk 2]
```diff
@@ -9,7 +9,6 @@ class Notification < Rpush::Client::Redis::Notification
 
           def time_to_live=(value)
             self.expiry = value
-            super
```
Comment (jhottenstein) [inline lib/rpush/client/redis/pushy/notification.rb]: time_to_live= is not defined on the modis model.

File: spec/unit/client/redis/apns/notification_spec.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+describe Rpush::Client::Redis::Apns::Notification do
+  it_behaves_like 'Rpush::Client::Apns::Notification'
+
+  it "should default the sound to 'default'" do
+    notification = described_class.new
+    expect(notification.sound).to eq('default')
+  end
+
+  # skipping these tests because data= for redis doesn't merge existing data
```
Comment (jhottenstein) [inline spec/unit/client/redis/apns/notification_spec.rb]: redis notification does not override the data= method to merge in the existing keys.  I'm not sure if we want that (there was a bug opened on that: #156)

File: spec/unit/client/redis/apns/notification_spec.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  end
+
+  xit 'does not overwrite the content-available flag when setting attributes for the device' do
+    notification.content_available = true
+    notification.data = { 'hi' => 'mom' }
+    expect(notification.as_json['aps']['content-available']).to eq 1
+    expect(notification.as_json['hi']).to eq 'mom'
+  end
+
+  # redis does not use alert_is_json - unclear if that is a bug or desired behavior
```
Comment (jhottenstein) [inline spec/unit/client/redis/apns/notification_spec.rb]: It doesn't seem like redis needs to use alert_is_json to handle storing the alert

File: .ruby-version [hunk 5]
```diff
@@ -1 +1 @@
-2.6.3
+2.6.4
```
Comment (aried3r) [inline .ruby-version]:
```suggestion
2.6.5
```

**Review by benlangfeld — APPROVED** [review]

**Pull request merged**